### PR TITLE
Rangefinder

### PR DESCRIPTION
**Pull request by kolbytn**
Added RangeFinderSensor and logic.

### CODE REVIEW

**Review by MaxRobinsonTheGreat — APPROVED** [review]
👍

File: Source/Holodeck/Sensors/Public/RangeFinderSensor.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+// Allows sensor parameters to be set programmatically from client.
+void URangeFinderSensor::ParseSensorParms(FString ParmsJson) {
+	Super::ParseSensorParms(ParmsJson);
+
+	TSharedPtr<FJsonObject> JsonParsed;
+	TSharedRef<TJsonReader<TCHAR>> JsonReader = TJsonReaderFactory<TCHAR>::Create(ParmsJson);
+	if (FJsonSerializer::Deserialize(JsonReader, JsonParsed)) {
+
+		if (JsonParsed->HasTypedField<EJson::Number>("LazerCount")) {
```
Comment (jaydenmilne) [inline Source/Holodeck/Sensors/Public/RangeFinderSensor.cpp]: Uhh, bad news, it's spelled "laser"

¯\\_(ツ)_/¯

File: Source/Holodeck/Sensors/Public/RangeFinderSensor.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		FCollisionQueryParams QueryParams = FCollisionQueryParams();
+		QueryParams.AddIgnoredActor(Parent);
+
+		FHitResult Hit = FHitResult();
+
+		bool TraceResult = GetWorld()->LineTraceSingleByChannel(Hit, start, end, ECollisionChannel::ECC_Visibility, QueryParams);
+		FloatBuffer[i] = (Hit.Distance != NULL ? Hit.Distance : LazerMaxDistance) / 100;  // centimeter to meters
+	
+		if (LazerDebug) {
+			DrawDebugLine(GetWorld(), start, end, FColor::Green, false, .01, ECC_WorldStatic, 1.f);
```
Comment (jaydenmilne) [inline Source/Holodeck/Sensors/Public/RangeFinderSensor.cpp]: Did you get rid of the ghosting problem?

File: Source/Holodeck/Sensors/Public/RangeFinderSensor.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+#pragma once
+
+#include "CoreMinimal.h"
+#include "HolodeckCore/Public/HolodeckSensor.h"
+#include "RangeFinderSensor.generated.h"
+
+/**
+ * URangeFinderSensor
+ * The RangeFinderSensor gets distances to first collision.
+ * By default the sensor retruns one number that represents the distance to
```
Comment (jaydenmilne) [inline Source/Holodeck/Sensors/Public/RangeFinderSensor.h]: s/retruns/returns

**Review by jaydenmilne — CHANGES_REQUESTED** [review]

**Review by daniekpo — APPROVED** [review]
Looks good to me apart from the spelling mistakes @jaydenmilne  suggested.

File: Source/Holodeck/Sensors/Public/RangeFinderSensor.cpp (hunk 1 above)
Comment (kolbytn) [reply Source/Holodeck/Sensors/Public/RangeFinderSensor.cpp]: Well it should be spelled lazer.... Lazer is much better...

File: Source/Holodeck/Sensors/Public/RangeFinderSensor.cpp (hunk 2 above)
Comment (kolbytn) [reply Source/Holodeck/Sensors/Public/RangeFinderSensor.cpp]: Yes. Shortening the duration time significantly solved the problem.

**Review by jaydenmilne — APPROVED** [review]

**Pull request merged**